src/ILLink.RoslynAnalyzer/RequiresAssemblyFilesAnalyzer.cs

```diff
@@ -94,10 +56,10 @@ protected override ImmutableArray<ISymbol> GetSpecialIncompatibleMembers (Compil
 
 			var assemblyType = compilation.GetTypeByMetadataName ("System.Reflection.Assembly");
 			if (assemblyType != null) {
-				// properties
+				// Properties
 				ImmutableArrayOperations.AddIfNotNull (dangerousPatternsBuilder, ImmutableArrayOperations.TryGetSingleSymbol<IPropertySymbol> (assemblyType.GetMembers ("Location")));
 
-				// methods
+				// Methods
```

Comment:
Maybe its a good time to delete the CodeBase and EscapeCodeBase special handling? The runtime is already annotated

Reply:
Nice. Shouldn't we remove all special cases? `GetFile`, `GetFiles` and `Location` are also annotated on the runtime, so I think we could get rid of this method.

Reply:
I'm up for it, but now that I think more, all of this would be a breaking change for people that had suppressed IL3000 and IL3001 and now will start having IL3002. On the other hand, it will even simplify the current analyzer code we would only generate IL3002 and IL3003. But I'm not very good at dimensioning if it's worth it the break, so I will leave @agocke to comment on that

Reply:
Don't we have a customized message for `Location`? Using the attribute for `GetFile` and `GetFiles` seems fine, but `Assembly.Location` seems like it might be worth a special message, if we can give a good one.

Reply:
Well, we control the message field in RAF, is that enough? Or we can change the Documentation URLs for IL3000 and IL3001 to not show the warning code (since they wouldn't be generated anymore) and pass them in RAF as URLs

Reply:
By the way, I gave a quick look at the runtime and in System.Reflection.Assembly the GetFile/GetFiles/Location are not annotated with RAF

Reply:
> By the way, I gave a quick look at the runtime and in System.Reflection.Assembly the GetFile/GetFiles/Location are not annotated with RAF

My bad. Yeah, just saw, some of these have RUC on them.

Reply:
Having said that we could annotate the runtime if we want